homeassistant/helpers/trigger.py

```diff
@@ -49,12 +49,29 @@
 )
 
 
+@dataclass
+class Trigger:
+    """Trigger descriptor data class."""
+
+    async_validate_trigger_config: Callable[
+        [HomeAssistant, ConfigType],
+        Coroutine[Any, Any, ConfigType],
+    ]
+    async_attach_trigger: Callable[
+        [HomeAssistant, ConfigType, TriggerActionType, TriggerInfo],
+        Coroutine[Any, Any, CALLBACK_TYPE],
+    ]
+
+
 class TriggerProtocol(Protocol):
     """Define the format of trigger modules.
 
-    Each module must define either TRIGGER_SCHEMA or async_validate_trigger_config.
+    New implementations should only implement async_get_triggers.
```

Comment:
this now seems to contradict this: https://github.com/home-assistant/core/pull/144742#discussion_r2085534014
personally, I prefer the method approach, but isn't the task spawn for a simple validation a concern here?

Reply:
I did a github code search, and I don't find anything implementing condition platforms outside of Home Assistant Core. I think we can remove the support for `CONDITION_SCHEMA` from condition platforms.

Reply:
It was added already in PR #88509, but not used.